src/Lumper.Lib/BSP/BspFile.cs

```diff
@@ -86,32 +287,30 @@ public void Save(Stream stream)
         };
 
         if (typeMap.ContainsKey(typeof(T)))
-        {
             return (T)Lumps[typeMap[typeof(T)]];
-        }
+
         IEnumerable<KeyValuePair<BspLumpType, Lump<BspLumpType>>> tLumps = Lumps.Where(x => x.Value.GetType() == typeof(T));
         return (T)tLumps.Select(x => x.Value).First();
     }
 
     public Lump<BspLumpType> GetLump(BspLumpType lumpType) => Lumps[lumpType];
 
-    public void ToJson(bool sortLumps,
-        bool sortProperties,
-        bool ignoreOffset)
+    // I tried to refactor a bunch of code to use URIs everywhere but was a big hassle and Uri ctor apparently
+    // doesn't handle escaped stuff well... just sticking with strings
+    private static string GetUnescapedFilePathString(string path) => Uri.UnescapeDataString(Path.GetFullPath(path));
```

Comment:
maybe FileInfo would help? don't know and I don't mind the strings

Reply:
Was messing with FileInfo a bunch and wasn't confident in it. Strings aren't too bad, and we already know this works on Linux so let's stick with it.